calcom/cal.com
# allow users to set a new password if they have signed up with Google or SAML provider.

## What does this PR do?

This PR add the support for setting a new password for users who have signed up with Google or SAML provider. This allow the same user to login with email/password in the future if they want to. The users can set their new password from `/settings/security/password` page. 

We should hide the `Old Password` field for these users for the first time. If the user has already set a password, we show the `Old Password` field (PENDING)

Since the `user` object doesn't have the password field, at the moment frontend have no idea if the user has already set a password or not. Either we should create a new API or update the `user` to return a flag to indicate if the user has already set a password or not.



**Environment**: Staging(main branch) / Production

## Type of change




## How should this be tested?



- Sign up with Google or SAML provider
- Go to `/settings/security/password` page
- You should see the `New Password` field and `Old Password` field should be hidden
- Set a new password
- Logout and login with email/password
- It should work
